punchcut/docs/capi.md

```diff
@@ -0,0 +1,154 @@
+# C API for punchcut
```

Comment:
Would it make sense to split this out from the pull request and discuss separately?

Reply:
I don't think I meant to push that to the branch for this PR. Per our discussion, we need something more sophisticated so I'm happy to remove the file and punt on this.

Reply:
Removed from this PR and stored at https://gist.github.com/dfrg/f1b9022541e66b881b6e82fd2f9969b1 for future reference.